packages/manager/modules/carbon-calculator/src/dashboard/carbon-footprint/controller.js

```diff
@@ -0,0 +1,10 @@
+export default class CarbonFootprintCtrl {
+  /* @ngInject */
+  constructor(carbonFootprintService) {
+    this.carbonFootprintService = carbonFootprintService;
+  }
+
+  $onInit() {
+    this.previousMonth = this.carbonFootprintService.constructor.computePreviousMonth();
```

Comment:
Maybe you could create this.previousMonth directly in constructor or if computePreviousMonth() is use only one time in application put this fonction directly in this controller ?

Reply:
The service will contain the API part so I prefer keep it but ok to use constructor instead of ```$onInit```.
But for me, constructor is made to create objects and ```$onInit``` is mode for initialize values

Reply:
fixed

Reply:
@ghyenne . To let you know, that fix created a code smells on sonar